src/Microsoft.Health.SqlServer/Features/Health/SqlServerHealthCheck.cs

```diff
@@ -52,18 +54,31 @@ public override async Task<HealthCheckResult> CheckStorageHealthAsync(Cancellati
 
             return HealthCheckResult.Healthy("Successfully connected.");
         }
-        catch (SqlException e) when (e.IsCMKError())
+        catch (HttpRequestException httpEx) when (IsInvalidAccess(httpEx))
+        {
+            HealthStatusReason reason = HealthStatusReason.DataStoreConnectionDegraded;
+
+            return new HealthCheckResult(
+                HealthStatus.Degraded,
+                DegradedDescription,
+                httpEx,
+                new Dictionary<string, object> { { "Reason", reason.ToString() } });
+        }
+        catch (SqlException sqlEx) when (sqlEx.IsCMKError())
         {
             // Error 40925: "Can not connect to the database in its current state". This error can be for various DB states (recovering, inacessible) but we assume that our DB will only hit this for Inaccessible state
-            HealthStatusReason reason = e.Number is SqlErrorCodes.CannotConnectToDBInCurrentState
+            HealthStatusReason reason = sqlEx.Number is SqlErrorCodes.CannotConnectToDBInCurrentState
                 ? HealthStatusReason.DataStoreStateDegraded
                 : HealthStatusReason.CustomerManagedKeyAccessLost;
 
             return new HealthCheckResult(
                 HealthStatus.Degraded,
                 DegradedDescription,
-                e,
+                sqlEx,
                 new Dictionary<string, object> { { "Reason", reason.ToString() } });
         }
     }
+
+    private static bool IsInvalidAccess(HttpRequestException exception)
```

Comment:
Should we put this in a common location? So its not duplicated across here + `SqlServerSchemaDataStore`